src/main/java/io/jenkins/plugins/jobrestrictions/util/ClassSelector.java

```diff
@@ -89,7 +90,9 @@ public String getDisplayName() {
             return "N/A";
         }
 
+        @RequirePOST
         public FormValidation doCheckSelectedClass(final @QueryParameter String selectedClass) {
+            Jenkins.get().checkPermission(Jenkins.ADMINISTER);
```

Comment:
I would also suppress here the security warning and remove the permission check.